packages/adapter-node/index.js

```diff
@@ -13,7 +13,16 @@ const files = fileURLToPath(new URL('./files', import.meta.url));
 export default function ({
 	out = 'build',
 	precompress,
-	env: { path: path_env = 'SOCKET_PATH', host: host_env = 'HOST', port: port_env = 'PORT' } = {}
+	env: {
+		path: path_env = 'SOCKET_PATH',
+		host: host_env = 'HOST',
+		port: port_env = 'PORT',
+		base: base_env,
+		headers: {
+			protocol: protocol_header_env = 'PROTOCOL_HEADER',
+			host: host_header_env = 'HOST_HEADER'
+		}
+	} = {}
```

Comment:
@Rich-Harris , you probably meant

```suggestion
	} = {headers:{}}
```

otherwise you get a `Cannot read properties of undefined (reading 'protocol')` when env is not specified.

Reply:
This is one of the things I'm attempting to address in #3423.